x-pack/test/api_integration/apis/ml/results/get_category_definition.ts

```diff
@@ -0,0 +1,184 @@
+      await spacesService.create({ id: idSpace2, name: 'space_two', disabledFeatures: [] });
+
+      await ml.api.createAndRunAnomalyDetectionLookbackJob(
+        // @ts-expect-error not full interface
+        testJobConfig,
+        testDatafeedConfig,
+        idSpace1
+      );
+    });
+
+    after(async () => {
+      await spacesService.delete(idSpace1);
+      await spacesService.delete(idSpace2);
+      await ml.testResources.deleteIndexPatternByTitle('ft_module_sample_logs');
+      await ml.api.cleanMlIndices();
+    });
+
+    it('should produce the correct category for the job', async () => {
+      const resp = await getCategoryDefinition(
+        jobIdSpace1,
+        expectedCategoryDefinition.categoryId,
+        USER.ML_POWERUSER,
+        200,
+        idSpace1
+      );
+
+      expect(resp.categoryId).to.eql(
+        expectedCategoryDefinition.categoryId,
+        `categoryId should be ${expectedCategoryDefinition.categoryId} (got ${resp.categoryId})`
+      );
+      expect(resp.examples.length).to.eql(
+        expectedCategoryDefinition.examples.length,
+        `examples list length should be ${expectedCategoryDefinition.examples.length} (got ${resp.examples.length})`
+      );
+      expect(resp.terms.length).to.be.greaterThan(
+        0,
+        `terms string length should be greater than 0 (got ${resp.terms.length})`
+      );
+    });
+
+    it('should not produce the correct category for the job', async () => {
+      const resp = await getCategoryDefinition(jobIdSpace1, '2', USER.ML_POWERUSER, 200, idSpace1);
+
+      expect(resp.categoryId).to.not.eql(
+        expectedCategoryDefinition.categoryId,
+        `categoryId should not be ${expectedCategoryDefinition.categoryId} (got ${resp.categoryId})`
+      );
+    });
+
+    it('should not produce the correct category for the job in the wrong space', async () => {
+      await getCategoryDefinition(
+        jobIdSpace1,
+        expectedCategoryDefinition.categoryId,
+        USER.ML_POWERUSER,
+        404,
+        idSpace2
+      );
+    });
+
+    it('should produce the correct category for ml viewer user', async () => {
+      const resp = await getCategoryDefinition(
+        jobIdSpace1,
+        expectedCategoryDefinition.categoryId,
+        USER.ML_VIEWER,
+        200,
+        idSpace1
+      );
+
+      expect(resp.categoryId).to.eql(
+        expectedCategoryDefinition.categoryId,
+        `categoryId should be ${expectedCategoryDefinition.categoryId} (got ${resp.categoryId})`
+      );
+      expect(resp.examples.length).to.eql(
+        expectedCategoryDefinition.examples.length,
+        `examples list length should be ${expectedCategoryDefinition.examples.length} (got ${resp.examples.length})`
+      );
+      expect(resp.terms.length).to.be.greaterThan(
+        0,
+        `terms string length should be greater than 0 (got ${resp.terms.length})`
+      );
```

Comment:
See comment above: we could add a check for `regex` here.

Reply:
Updated in https://github.com/elastic/kibana/pull/124710/commits/7634363c0ce8420f1ce649f17012731ecce30cae